### PR TITLE
fix: Various edits

### PR DESCRIPTION
**Pull request by robin-aws**
Credit to @RustanLeino and @cpitclaudel for most of these suggestions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: instructions/steps.md [hunk 1]
```diff
@@ -304,26 +281,24 @@ albeit not always immutable.
...
 
 Let's create one!
 
 <!-- !test check AwsKmsArn -->
 ```dafny
 
-  type AwsKmsArn = arn: AwsArn
-  | AwsKmsArn?(arn)
-  witness *
+  type AwsKmsArn = arn: AwsArn | AwsKmsArn?(arn) witness *
```
Comment (seebees) [inline instructions/steps.md:301]: sigh, when the formatting gods come and smite you... :)

File: instructions/steps.md [hunk 2]
```diff
@@ -384,9 +357,8 @@ Copy over these naive signatures.
 `function` is mostly what you expect.
 However, `function`s in Dafny are more restrictive
 than you are probably used to.
-They are syntactic sugar for expressions.
 They are more like mathematical functions: 100% deterministic.
-The can't mutate, use loops, or create objects on the heap.
+The can't mutate, use loops, or create objects.
```
Comment (seebees) [inline instructions/steps.md:361]: Hum, I'll leave it, but we do create `datatype` in functions.
And the distention may come up.
Not blocking

File: instructions/steps.md [hunk 3]
```diff
@@ -532,7 +506,7 @@ One for the `Success` and the other for `Failure`[^monad]
 [^monad]: If this sounds to you like a monad,
     then congratulations it is pretty close.
     If you have no idea what a monad is,
-    then congratulations you are one of the lucky 10,000!
+    then [congratulations](https://en.wikipedia.org/wiki/Monad_(functional_programming)) you are one of the [lucky 10,000](https://xkcd.com/1053/)!
```
Comment (seebees) [inline instructions/steps.md:509]: thanks

**Review by seebees — APPROVED** [review]

File: instructions/steps.md [hunk 4]
```diff
@@ -66,32 +68,9 @@ module {:options "-functionSyntax:4"} AwsKmsArnParsing {
 You will also see the Dafny extension automatically download Dafny for you,
 if this is the first time you've opened a Dafny source file.
 
-Let's go over a few Dafny basics:
-
-1. `include` is how Dafny includes other files.
-The file `include.dfy` is a helper file we added for you.
+There's no need to understand the details of what's already in the file,
+but if you are keen you can read more [here](#notes-on-the-initial-state).
```
[the remark this answers is not in the document]
Comment (robin-aws) [reply instructions/steps.md:72]: I considered that but it's a lot of text and I figured you would lose the formatting.

File: instructions/steps.md (hunk 1 above)
Comment (robin-aws) [reply instructions/steps.md:301]: This was a common source of feedback from multiple reviewers and I agree with them. :)

Personally if you REALLY want multiple lines, I'd at least want to indent subsequent lines. But even then there I worry about `| ...` looking too much like a constructor for a datatype.

**Pull request merged**